pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -467,28 +473,55 @@ func (r *ReferenceAdapter) deleteCredentials() error {
 	return nil
 }
 
+type EnsureAzResult int
+
+const (
+	EnsureAzResultNotReady EnsureAzResult = iota
+	EnsureAzResultModified
+	EnsureAzResultNoChange
+)
+
 // ensureAvailibilityZonesSet sets the az in the projectclaim spec if necessary
 // returns true if the project claim has been modified
-func (r *ReferenceAdapter) ensureClaimAvailibilityZonesSet() (bool, error) {
+func (r *ReferenceAdapter) ensureClaimAvailibilityZonesSet() (EnsureAzResult, error) {
 	if len(r.ProjectClaim.Spec.AvailibilityZones) > 0 {
-		return false, nil
+		return EnsureAzResultNoChange, nil
 	}
 
 	zones, err := r.gcpClient.ListAvilibilityZones(r.ProjectReference.Spec.GCPProjectID, r.ProjectClaim.Spec.Region)
 	if err != nil {
-		return false, err
+		return r.handleAvailabilityZonesError(err)
 	}
+	conditions := &r.ProjectReference.Status.Conditions
+	r.conditionManager.SetCondition(conditions, gcpv1alpha1.ConditionComputeApiReady, corev1.ConditionTrue, "QueryAvailabilityZonesSucceeded", "ComputeAPI ready, successfully queried availability zones")
```

Comment:
is there a global place to find constants is this repo? I searched but it seems the state `QueryAvailabilityZonesSucceeded` is only saved where the code is called, I expected to have a consts file with all of them

Reply:
as long as we don't need it in another package, I don't see much reason to move it. What do you think would be the benefit of moving it?

Reply:
to have all of the statuses in one place so we can identify where each state is used

Reply:
As long as this is not used outside the module, I prefer to keep it in this file. That being said, I think it should be private.